shared-module/audiofilters/Filter.c

```diff
@@ -60,11 +60,11 @@ void common_hal_audiofilters_filter_construct(audiofilters_filter_obj_t *self,
 
     // The below section sets up the effect's starting values.
 
-    if (filter == MP_OBJ_NULL) {
-        filter = mp_const_none;
+    if (filters == MP_OBJ_NULL) {
+        filters = mp_obj_new_list(0, NULL);
```

Comment:
also you don't need to create an empty list, you can require that the empty tuple (that can also be the default constructor argument) is used if zero filters are actually intended.

Reply:
> call `common_hal_audiofilters_filter_set_filters` here instead of duplicating code

I'm good with this. I haven't seen to many calls to `common_hal_...` functions in other implementations of `common_hal_..._construct(..)`, so I wasn't sure if that was a bad practice.

> also you don't need to create an empty list, you can require that the empty tuple (that can also be the default constructor argument) is used if zero filters are actually intended.

I think I'm going to switch it out to default to `mp_const_none` instead. I like the idea of turning off the filter without rearranging the audio signal path, by setting `filters = None`. In that case, `common_hal_audiofilters_filter_set_filters` will set `filter_states = NULL` and `audiofilters_filter_get_buffer` will simply pass the sample through.